qt/widgets/common/src/AddWorkspaceMultiDialog.cpp

```diff
@@ -0,0 +1,86 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidQtWidgets/Common/AddWorkspaceMultiDialog.h"
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidQtWidgets/Common/ParseKeyValueString.h"
+#include <QFileInfo>
+
+#include <boost/optional.hpp>
+#include <utility>
+
+namespace {
+using namespace Mantid::API;
+
+void loadFile(const QString &filename) {
+  QFileInfo fileinfo(filename);
+
+  auto loader = AlgorithmManager::Instance().createUnmanaged("Load");
+  loader->initialize();
+  loader->setProperty("Filename", filename.toStdString());
+  loader->setProperty("OutputWorkspace", fileinfo.baseName().toStdString());
+  loader->execute();
+
+  return;
+}
+} // namespace
+
+namespace MantidQt::MantidWidgets {
+
+AddWorkspaceMultiDialog::AddWorkspaceMultiDialog(QWidget *parent) : QDialog(parent) {
+  m_uiForm.setupUi(this);
+
+  connect(m_uiForm.pbSelAll, SIGNAL(clicked()), this, SLOT(selectAllSpectra()));
+  connect(m_uiForm.pbResetDefault, SIGNAL(clicked()), this, SLOT(updateSelectedSpectra()));
+  connect(m_uiForm.dsInputFiles, SIGNAL(filesFound()), this, SLOT(handleFilesFound()));
+  connect(m_uiForm.pbUnify, SIGNAL(clicked()), this, SLOT(unifyRange()));
+  connect(m_uiForm.pbAdd, SIGNAL(clicked()), this, SLOT(emitAddData()));
+  connect(m_uiForm.pbClose, SIGNAL(clicked()), this, SLOT(close()));
+}
+std::string AddWorkspaceMultiDialog::workspaceName() const { return std::string(""); }
```

Comment:
It looks like this function should always be unused. Can we throw a `std::logic_error` to say that this method is not implemented and should not be used for an `AddWorkspaceMultiDialog`?

Reply:
The dialog window is behaving weirdly when adding the throw statement (it crashes the workbench).

Reply:
Thats odd, is it possible the `AddWorkspaceMultiDialog::workspaceName()` function is being called from somewhere? When you set a breakpoint does it get entered?